src/Symfony/Component/Security/Core/Authentication/Token/PreAuthenticatedToken.php

```diff
@@ -24,20 +24,28 @@ class PreAuthenticatedToken extends AbstractToken
     private $firewallName;
 
     /**
-     * @param string|\Stringable|UserInterface $user
-     * @param mixed                            $credentials
-     * @param string[]                         $roles
+     * @param UserInterface $user
+     * @param string        $firewallName
+     * @param string[]      $roles
      */
-    public function __construct($user, $credentials, string $firewallName, array $roles = [])
+    public function __construct($user, /*string*/ $firewallName, /*array*/ $roles = [])
```

Comment:
Hi guys, I'm not sure how to use `UsernamePasswordToken` without `$credentials` as this change is not explained in documentation: https://symfony.com/doc/current/components/security/authentication.html

Reply:
You probably get faster answers if you use one of the support channels: https://symfony.com/support (e.g. GitHub discussions) Not a lot of people are watching already merged PRs.

The security component is refactored during Symfony 5. Tokens now only represent authenticated tokens/sessions, so they no longer have credentials. The "pre authentication tokens" are now called passports.

We haven't updated the components docs yet, but the framework docs are upgraded to the new system. Hopefully, you'll be able to figure stuff out using e.g. https://symfony.com/doc/current/security/custom_authenticators.html